src/dev/server.ts

```diff
@@ -173,7 +173,14 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
     eventHandler(async (event) => {
       await reloadPromise;
       const address = currentWorker && currentWorker.address;
-      if (!address || (address.socketPath && !existsSync(address.socketPath))) {
+      if (
+        !address ||
+        (address.socketPath &&
+          !(await fsp.access(address.socketPath).then(
```

Comment:
Maybe we can try sync version to avoid a stalled status? (access can be not available in next async think we call actual error handler)..

Reply:
Do you mean using https://nodejs.org/api/fs.html#fsaccesssyncpath-mode instead of the promise version ?
This change should fix https://github.com/nuxt/nuxt/issues/15164

Reply:
@Hebilicious 

You cannot just use `accessSync` since it may throw an error:

> If any of the accessibility checks fail, an Error will be thrown. Otherwise, the method will return undefined. (https://nodejs.org/api/fs.html#fsaccesssyncpath-mode)

This causes a dev server to be completely down. To avoid this we should wrap `accesSync` in try-catch. 

Even with that done, the problem does not seem to be fixed. The server cannot handle 200 requests in my case:

```javascript
for (let i = 0; i < 200; i++) {
  fetch('http://localhost:3000/api/test')
}
```

```typescript
// server/api/test.get.ts

export default eventHandler(() => {
  return '123098'
})
```

> POST http://localhost:3000/api/test 503 (Server Unavailable)

I've done more stressful tests with [autocannon](https://www.npmjs.com/package/autocannon). The results:
~ 81k requests in 5.05s, 1.95 MB read
~ 64k errors